### DISCUSSION TITLE
在ts中使用jsx会报这个错误应该怎么解决呢？

### DISCUSSION THREAD

**Discussion by Mywbj** [eslint/eslint#16634] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/67814295/206620108-f03d625f-f554-449f-9a45-c1bffa468fe2.png)
在ts中使用jsx会报这个错误应该怎么解决呢？
![image](https://user-images.githubusercontent.com/67814295/206620072-701bb8f6-d85d-4a22-9868-fe23087e8b47.png)

**Comment by sam3k** [comment, marked as answer]
Hi @Mywbj, usually this error occurs when the file is not a `.tsx` format. So simply renaming from `.ts` to `.tsx` should fix this issue. If it does not, could you provide a reproducible example?

Once you have renamed the file the error should go away if the `react/self-closing-comp` rule is properly set:

```javascript
"react/self-closing-comp": ["error", {
  "component": true,
  "html": true
}]
```

**Reply by Mywbj** [reply]
Finally, I changed it to. tsx. It may be that I was too rigid before. It may also be for the sake of specification that this error is reported. Writing jsx code should be written in jsx. Thank you for your reply. Thank you very much
